ocaml-lsp-server/src/document.ml

```diff
@@ -436,13 +440,18 @@ let get_impl_intf_counterparts m uri =
     in
     match Syntax.of_fname fname with
     | Dune | Cram -> []
+    (* TODO: Unsure about this, keeping it empty for now *)
+    | Mlx ->
+      (match kind with
+       | Intf -> [ re; ml; mly; mll ]
```

Comment:
```suggestion
       | Intf -> [ re; ml; mly; mll; mlx ]
```
as we want to be able to switch to .mlx from .mli if it is present

Reply:
But here we are matching on an "Mlx Intf", which is not an `mli` file, right ? I am not sure that case happens at all. But nothing really bad could happen from having too many of such rules...

Reply:
looking at the surrounding code, I thought the `kind` belongs to the origin, so we are on interface file and want to switch to the implementation, which can be .re, .ml, .mly, mll, and now .mlx